packages/design-system-web/src/components/geov-yasgui-data-validation/geov-yasgui-data-validation.tsx

```diff
@@ -0,0 +1,96 @@
+import { Component, Host, Prop, h, EventEmitter, Event } from '@stencil/core';
+import type { Parser } from '@triply/yasr';
+
+interface ExpectedKey {
+  name: string;
+  required: boolean;
+  datatype: string;
+  customValidator?: (val: Parser.BindingValue) => Set<string> | undefined;
+}
+/**
+ * The component has to validate input data and emit the validation result (valid/invalid).
+ *
+ * It requires two inputs: 'data' (the input data to be validated) and 'expectedKeys' (an array of expected keys).
+ *
+ * The component emits a custom event named "validationCompleted" with a boolean value:
+ *   - true if the validation passes,
+ *   - false if there are validation errors.
+ *
+ * In case of invalid data, it has to display understandable warnings/error messages.
+ */
```

Comment:
```suggestion
/**
 * The component validates data and emits the validation result (valid/invalid).
 *
 * It has two required inputs: `data` (the input data to be validated) and 'expectedKeys' (an array of expected keys containing the validation rules).
 *
 * The component emits a custom event named "validationCompleted" with a boolean value:
 *   - true if the validation passes,
 *   - false if there are validation errors.
 *
 * In case of invalid data, it displays understandable warnings/error messages.
 */
```